rmf_task/src/rmf_task/agv/TaskPlanner.cpp

```diff
@@ -742,44 +764,21 @@ class TaskPlanner::Implementation
 
   double compute_h(const Node& node, const rmf_traffic::Time time_now)
   {
-    std::vector<double> initial_queue_values;
-    initial_queue_values.resize(
-          node.assigned_tasks.size(), std::numeric_limits<double>::infinity());
+    std::vector<double> initial_queue_values(
+      node.assigned_tasks.size(), 0.0);
 
-    for (const auto& u : node.unassigned_tasks)
+    // Determine earliest possible deployment time for each agent
+    for(size_t i = 0; i < node.assigned_tasks.size(); ++i)
     {
-      // We subtract the invariant duration here because otherwise its
-      // contribution to the cost estimate will be duplicated in the next section,
-      // which could result in an overestimate.
-      const rmf_traffic::Time variant_time =
-          u.second.candidates.best_finish_time()
-          - u.second.request->invariant_duration();
-      const double variant_value =
-        rmf_traffic::time::to_seconds(variant_time - time_now);
-
-      const auto& range = u.second.candidates.best_candidates();
-      for (auto it = range.begin; it != range.end; ++it)
+      if (node.assigned_tasks[i].empty())
       {
-        const std::size_t candidate = it->second.candidate;
-        if (variant_value < initial_queue_values[candidate])
-          initial_queue_values[candidate] = variant_value;
+        initial_queue_values[i] = rmf_traffic::time::to_seconds(
+          time_now.time_since_epoch());
       }
-    }
-
-    for (std::size_t i=0; i < initial_queue_values.size(); ++i)
-    {
-      auto& value = initial_queue_values[i];
-      if (std::isinf(value))
+      else
       {
-        // Clear out any infinity placeholders. Those candidates simply don't have
-        // any unassigned tasks that want to use it.
-        const auto& assignments = node.assigned_tasks[i];
-        if (assignments.empty())
-          value = 0.0;
-        else
-          value =
-            rmf_traffic::time::to_seconds(
-              assignments.back().state().finish_time() - time_now);
+        initial_queue_values[i] = rmf_traffic::time::to_seconds(
```

Comment:
It seems that we are considering the finish times of charging tasks as well here while ignoring it in the `compute_g()` function. Should we be ignoring them here as well?

Reply:
Here we determine the earliest possible time the agent can begin any currently unassigned task. In the `InvariantHeuristicQueue` class, we use this value to determine the earliest time an unassigned task may finish (by adding the invariant costs leading up to it in the queue), then subtract from this finish time its specified earliest start time, mimicking `g(n)`. So we're only including the charging task when determining the earliest time the agent can begin a task, but not separately adding it to the queue (the initial queue value is a bit special in that sense).

Reply:
Ah right I overlooked that. Thanks for the clarification.